src/imitation/scripts/analyze.py

```diff
@@ -74,62 +123,170 @@ def gather_tb_directories(
+
+# verbosity 0
+table_verbosity_mapping.append(
+    {
+        "algo",
+        "env_name",
+        "expert_return_summary",
+        "imit_return_summary",
+    }
+)
+
+# verbosity 1
+table_verbosity_mapping.append(table_verbosity_mapping[-1] | {"n_expert_demos"})
+
+# verbosity 2
+table_verbosity_mapping.append(
+    table_verbosity_mapping[-1]
+    | {"status", "imit_expert_ratio", "exp_command", "run_name"}
+)
+
+
+def _get_table_entry_fns_subset(table_verbosity: int) -> sd_to_table_entry_type:
+    assert table_verbosity >= 0
+    if table_verbosity >= len(table_verbosity_mapping):
+        return table_entry_fns
+    else:
+        keys_subset = table_verbosity_mapping[table_verbosity]
+        result = OrderedDict()
+        for k, v in table_entry_fns.items():
+            if k not in keys_subset:
+                continue
+            result[k] = v
+        return result
+
+
 @analysis_ex.command
 def analyze_imitation(
-    source_dir: str,
-    run_name: Optional[str],
-    env_name: Optional[str],
-    skip_failed_runs: bool,
     csv_output_path: Optional[str],
-    verbose: bool,
+    tex_output_path: Optional[str],
+    print_table: bool,
+    table_verbosity: int,
 ) -> pd.DataFrame:
     """Parse Sacred logs and generate a DataFrame for imitation learning results.
 
+    This function calls the helper `_gather_sacred_dicts`, which captures its arguments
+    automatically via Sacred. Provide those arguments to select which Sacred
+    results to parse.
+
     Args:
-      source_dir: A directory containing Sacred FileObserver subdirectories
-        associated with the `train_adversarial` Sacred script. Behavior is
-        undefined if there are Sacred subdirectories associated with other
-        scripts.
-      run_name: If provided, then only analyze results from Sacred directories
-        associated with this run name. `run_name` is compared against the
-        "experiment.name" key in `run.json`.
-      skip_failed_runs: If True, then filter out runs where the status is FAILED.
-      csv_output_path: If provided, then save a CSV output file to this path.
-      verbose: If True, then print the dataframe.
+        csv_output_path: If provided, then save a CSV output file to this path.
+        tex_output_path: If provided, then save a LaTeX-format table to this path.
+        print_table: If True, then print the dataframe to stdout.
+        table_verbosity: Increasing levels of verbosity, from 0 to 2, increase the
+            number of columns in the table.
 
     Returns:
-      A list of dictionaries used to generate the analysis DataFrame.
+        The DataFrame generated from the Sacred logs.
     """
-    sacred_dicts = _get_sacred_dicts(source_dir, run_name, env_name, skip_failed_runs)
+    table_entry_fns_subset = _get_table_entry_fns_subset(table_verbosity)
 
     rows = []
-    for sd in sacred_dicts:
+    for sd in _gather_sacred_dicts():
         row = OrderedDict()
         rows.append(row)
```

Comment:
Nit: seems more natural to have this at the end of the `for` loop, once `row` has been fully set, although I know it doesn't actually matter since `row` is mutable and points at the same object throughout.